### ISSUE TITLE
Why "The script at 'https://cdn.jsdelivr.net/pyodide/v0.20.0/full/pyodide.js' failed to load."??

### ISSUE THREAD

**Issue by cosmosanalytics** [pyodide/pyodide#2687]
![image](https://user-images.githubusercontent.com/100080151/173191487-5dc390c8-f612-47a4-8b7e-149b1728f6c4.png)

I am trying to run https://jupyterlite.github.io/demo/lab/index.html in my company's computer, and getting the following error message in the console. But it is working in my home computer. How to make it work in my company's computer?

Failed to load resource: the server responded with a status of 404 (Not Found)
243.74743fa9e30be52e69d5.js:2 Deprecated include of L.Mixin.Events: this property will be removed in future releases, please inherit from L.Evented instead. Error
at https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/243.74743fa9e30be52e69d5.js:2:3143
at C.extend (https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/243.74743fa9e30be52e69d5.js:2:3162)
at 665 (https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/665.630b45b3f0d4998b678f.js:1:4763)
at j (https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/remoteEntry.6375e54406c677384240.js:1:795)
at https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/remoteEntry.6375e54406c677384240.js:1:3895
at w. (https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/remoteEntry.6375e54406c677384240.js:1:9878)
at j (https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/remoteEntry.6375e54406c677384240.js:1:795)
at 8197 (https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/138.f05c7875d6293e411f40.js:1:60671)
at j (https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/remoteEntry.6375e54406c677384240.js:1:795)
at 4795 (https://jupyterlite.github.io/demo/extensions/jupyter-leaflet/static/138.f05c7875d6293e411f40.js:1:44434)
(anonymous) @ 243.74743fa9e30be52e69d5.js:2
index.js:73 JupyterLab extension jupyter-wren-syntax is activated!
eea6526d-bf73-4855-b2d2-2658d86fa7fb:1 Uncaught DOMException: Failed to execute 'importScripts' on 'WorkerGlobalScope': The script at 'https://cdn.jsdelivr.net/pyodide/v0.20.0/full/pyodide.js' failed to load.
at blob:https://jupyterlite.github.io/eea6526d-bf73-4855-b2d2-2658d86fa7fb:1:1
(anonymous) @ eea6526d-bf73-4855-b2d2-2658d86fa7fb:1
272.eb0d340d1eec40ac564f.js:1 Failed to fetch ipywidgets through the "jupyter.widget.control" comm channel, fallback to fetching individual model state. Reason: Control comm did not respond in time

**Comment by rth** [comment]
Can you have a look at the network tab to see if  `pyodide.js` was loaded? Also can you try the [Pyodide REPL](https://pyodide.org/en/stable/console.html) which would be easier to debug?

**Comment by cosmosanalytics** [comment]
I can't load Pyodide REPL??
![image](https://user-images.githubusercontent.com/100080151/173232893-f93e5f62-2fa1-489c-b8cd-d82049c8dd47.png)

**Comment by cosmosanalytics** [comment]
network tab shows pyodide.js 403
![image](https://user-images.githubusercontent.com/100080151/173232952-771e13f2-c16d-4954-90bf-5f42715c5208.png)

**Comment by rth** [comment]
Likely your corporate proxy blocking connections.

**Comment by cosmosanalytics** [comment]
is there a way to unblock it?

**Comment by ryanking13** [comment]
It seems like your company blocks the access to `cdn.jsdelivr.net` (my company blocks it too :/). I think you should ask your company to unblock  `cdn.jsdelivr.net`.

Note that you can also host your own version of Pyodide. In jupyterlite: https://jupyterlite.readthedocs.io/en/latest/howto/python/pyodide.html

**Comment by cosmosanalytics** [comment]
thanks, so I just download the build Pyodide from https://github.com/pyodide/pyodide/releases, and serve it in intranet as explained in https://pyodide.org/en/stable/usage/downloading-and-deploying.html, right?

**Comment by cosmosanalytics** [comment]
I have downloaded the build Pyodide, but I also need to download jquery, because they are also coming from cdn.jsdelivr.net, can I also download them somewhere, like pyodide.js?
![image](https://user-images.githubusercontent.com/100080151/173276736-8a5185ad-a844-4e26-9580-ece3bb5c51bd.png)

**Comment by ryanking13** [comment]
Oh yeah, we are loading jquery and jquery terminal from `cdn.jsdelivr.net` in the REPL...
So if you want to use Pyodide REPL (of course you can use Pyodide without REPL), you'll have to modify CDN urls inside `console.html`. You can download jquery and jquery terminal locally, or use other CDNs.

- https://releases.jquery.com/
- https://terminal.jcubic.pl/

**Comment by HamDev235** [comment]
I changed console.html as follows
![image](https://user-images.githubusercontent.com/102918374/173571599-05bcf7dc-94b5-487e-8b9e-d5b1b70c9877.png)
Still got the following error message
![image](https://user-images.githubusercontent.com/102918374/173571927-52b47325-320e-4665-8fa9-2ab59690827f.png)

**Comment by HamDev235** [comment]
is it possible to deliver the pyodide via https://cdnjs.com/? my company does not block https://cdnjs.com/.

**Comment by HamDev235** [comment]
Kindly ask if you can write console.html in terms of  https://cdnjs.com/. Thanks a lot!

**Comment by rth** [comment]
Feel free to make deployment on a different CDN if that works for you.  For us changing the CDN provider is a lot of work, and we won't do it unless there is evidence from a significant fraction of users that a different CDN is better. Also, we have a custom setup with JsDelivr and I don't think cdnjs has that functionality.

> Still got the following error message

That likely means that jquery terminal wasn't loaded correctly.

**Comment by HamDev235** [comment]
Thanks!
Now it is working locally.

**Comment by alexmojaki** [comment]
I just learned that cdn.jsdelivr.net is apparently not working in China at the moment: https://forum.files.gallery/d/134-china-vs-cdnjsdelivrnet

But replacing `cdn` with `fastly` seems to work, e.g. https://fastly.jsdelivr.net/pyodide/v0.20.0/full/pyodide.js

**Comment by rth** [comment]
Closing as it doesn't sound that there is much we can do about this on our side.

> seems to work, e.g. fastly.jsdelivr.net

Nice, I wasn't aware one could select the CDN provider in Jsdelivr manually (normally it uses a mixture of providers depending on the location). Maybe we should mention it in the docs.

**Comment by cosmosanalytics** [comment]
Thanks!

Now I am using fastly for all my cdn needs.
